test/lib/recordevents/event_info_store.go

```diff
@@ -276,11 +268,51 @@ func (ei *EventInfoStore) WaitMatchSourceData(source string, data string, minCou
 	return nil
 }
 
+// Deprecated: use AssertAtMost
 func (ei *EventInfoStore) AssertWaitMatchSourceData(tb testing.TB, source string, data string, minCount int, maxCount int) {
 	if err := ei.WaitMatchSourceData(source, data, minCount, maxCount); err != nil {
 		tb.Fatalf("Timeout waiting for source %q and data %q. It does not appear at least %d times in the event record pod %q: %v", source, data, minCount, ei.podName, err)
 	}
 }
 
-// Does the provided EventInfo match some criteria
-type EventInfoMatchFunc func(EventInfo) error
+// Assert that there are at least min number of matches of f.
+// This method fails the test if the assert is not fulfilled.
+func (ei *EventInfoStore) AssertAtLeast(min int, f EventInfoMatcher) []EventInfo {
+	events, err := ei.WaitAtLeastNMatch(f, min)
+	if err != nil {
+		ei.tb.Fatalf("Timeout waiting for at least %d matches. Error: %v", min, err)
+	}
+	return events
+}
+
+// Assert that there are at least min number of matches and at most max number of matches of f.
+// This method fails the test if the assert is not fulfilled.
+func (ei *EventInfoStore) AssertAtMost(min int, max int, f EventInfoMatcher) []EventInfo {
```

Comment:
Name here is a bit unintuitive when compared to AssertAtLeast. How about AssertInRange(... )?

Reply:
That sounds good too